### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by nikimaxim**
Hey. I corrected some shortcomings in the work of monitoring. If you need screenshots of the problems play problems and make screen.

### CODE REVIEW

File: discovery-scripts/windows/smartctl-disks-discovery.ps1 [hunk 1]
```diff
@@ -1,53 +1,58 @@
...
+    exit
 }
 
-
 $idx = 0
 $global_serials
 $smart_scanresults = & $smartctl "--scan-open"
-$smart_scanresults += & $smartctl "--scan-open" "-dnvme"
+$smart_nvme = & $smartctl "--scan-open" "-dnvme"
+# bug fix. more than one disk
```
Comment (v-zhuravlev) [inline discovery-scripts/windows/smartctl-disks-discovery.ps1]: please remove this comment in final PR

File: discovery-scripts/windows/smartctl-disks-discovery.ps1 [hunk 2]
```diff
@@ -1,53 +1,58 @@
-# VERSION = 1.5
+# VERSION = 1.6
```
Comment (v-zhuravlev) [inline discovery-scripts/windows/smartctl-disks-discovery.ps1]: VERSION in perl script and userparams file should be changed as well then.

Comment (nikimaxim) [reply discovery-scripts/windows/smartctl-disks-discovery.ps1]: correct

File: discovery-scripts/windows/smartctl-disks-discovery.ps1 (hunk 1 above)
Comment (nikimaxim) [reply discovery-scripts/windows/smartctl-disks-discovery.ps1]: correct

**Comment by v-zhuravlev** [comment]
> %ProgramFiles% for me on Windows 10. Can you check it again please?

Can you check this one please?

**Comment by nikimaxim** [comment]
> > %ProgramFiles% for me on Windows 10. Can you check it again please?
> 
> Can you check this one please?

I had to return the settings. and the problem reproduced. 
It does not occur on all Windows 10 PCs. 
The problem has been more than a year since we started moving to Window 10. And just now got around to fix

items all problem(example one)
Preprocessing failed for: ???।???????? ??????: \smartmontools\bin\smartctl.exe" -i -H -A -l error -l background /dev/sdb -...
1. Failed: cannot perform regular expression "Offline_Uncorrectable.+ ([0-9]+)" match for value of type "string": pattern does not match

**Comment by nikimaxim** [comment]
Will you accept pull requests?

**Comment by v-zhuravlev** [comment]
Sorry, It would be easier to accept them if changes are more atomic and in different PRs.

For instance, i'm not sure if changing to static "C:\Program files" path is a good idea, but I would love to accept first fix you made. 

As for the last fix you made, It would be great to have your comment what was wrong and what did you actually changed. 
Thank you for you understanding.

**Comment by nikimaxim** [comment]
> For instance, i'm not sure if changing to static "C:\Program files" path is a good idea, but I would love to accept first fix you made.

1. I don't understand what this variable is for. if it in x86 and x64 issues the same. "C:\Program Files"
2. The problem is in Windows 10!

**Comment by v-zhuravlev** [comment]
> > For instance, i'm not sure if changing to static "C:\Program files" path is a good idea, but I would love to accept first fix you made.
> 
> 1. I don't understand what this variable is for. if it in x86 and x64 issues the same. "C:\Program Files"
> 2. The problem is in Windows 10!

It is since it is possible that system is installed not on C:, but different partition. Yes, I have two machines running Windows 10 with zbx-smartctl with no issues using such macro. So before blindly changing this, It would be great to find the reason why it doesn't work for you. Which version of agent do you have? Is this the latest agent from Zabbix official msi package?

**Comment by nikimaxim** [comment]
> It is since it is possible that system is installed not on C:, but different partition. Yes, I have two machines running Windows 10 with zbx-smartctl with no issues using such macro. So before blindly changing this, It would be great to find the reason why it doesn't work for you. Which version of agent do you have? Is this the latest agent from Zabbix official msi package?

The problem was:
Zabbix agent 4.0, 4.2. take the archive
Windows 10 version is also different

**Pull request closed without merging**